plan/relations.go

```diff
@@ -745,6 +953,63 @@ func (ar *AggregateRel) ToProtoPlanRel() *proto.PlanRel {
 	}
 }
 
+func (ar *AggregateRel) GetInputs() []Rel {
+	return []Rel{ar.input}
+}
+
+func (ar *AggregateRel) Copy(newInputs ...Rel) (Rel, error) {
+	if len(newInputs) != 1 {
+		return nil, substraitgo.ErrInvalidInputCount
+	}
+	aggregate := *ar
+	aggregate.input = newInputs[0]
+	return &aggregate, nil
+}
+
+func (ar *AggregateRel) CopyWithExpressionRewrite(rewriteFunc func(expr.Expression) (expr.Expression, error), newInputs ...Rel) (Rel, error) {
+	if len(newInputs) != 1 {
+		return nil, substraitgo.ErrInvalidInputCount
+	}
+	newGroups := make([][]expr.Expression, len(ar.groups))
+	groupsAreEqual := true
+	for i, g := range ar.groups {
+		newGroup := make([]expr.Expression, len(g))
+		for j, e := range g {
+			newExpr, err := rewriteFunc(e)
+			if err != nil {
+				return nil, err
+			}
+			if newExpr != e {
+				groupsAreEqual = false
+			}
+			newGroup[j] = newExpr
+		}
+		newGroups[i] = newGroup
+	}
```

Comment:
Similar to the above we can simplify this:

```go
var (
    newGroups = make([][]expr.Expressions, len(ar.groups)
    groupsAreEqual = true
    err error
)

for i, g := range ar.groups {
    newGroups[i] = make([]expr.Expression, len(g))
    for j, e := range g {
        if newGroups[i][j], err = rewriteFunc(e); err != nil {
             return nil, err
        }
        groupsAreEqual = groupsAreEqual && newGroups[i][j] == e
    }
}
```

Something similar can be done for the measures loop below